js/components/graph/Sidebar.js

```diff
@@ -0,0 +1,182 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Focus from "./Focus";
+
+export default class Sidebar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      contentHidden: true,
+      focusDisabled: false,
+      focusActive: false,
+      graphActive: false,
+      graphName: "",
+      toggled: false
+    }
+  }
+
+  componentWillUpdate(prevProps) {
+    // Check to see if we have a graph from local storage on load
+    if (this.state.graphName.length == 0 && prevProps.graphName !== this.state.graphName) {
+      this.setState({ graphName: prevProps.graphName }, () => {
+        this.handleGraph();
+      });
+    }
+  }
+
+  handleGraph = graphName => {
+    if (graphName != undefined) {
+      this.props.getGraph(graphName);
+    } else {
+      graphName = this.state.graphName;
+    }
+    // Enable Focuses nav if CS graph is selected
+    if (graphName === "Computer Science") {
+      this.setState({
+        focusDisabled: false,
+      });
+    } else {
+      this.setState({
+        focusDisabled: true,
+      });
+    }
+  }
+
+  createGraphButtons = () => {
+    const graphButtons = [];
+    this.props.graphs.forEach((graph, i) => {
```

Comment:
This function can be simplified by using `map` instead of `forEach`. (The code should look like `return this.props.graphs.map(...)`)